modules/javafx.controls/src/test/java/test/javafx/scene/control/LabeledTruncatedTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.javafx.scene.control;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import javafx.scene.control.Label;
+import javafx.scene.control.Labeled;
+import javafx.scene.control.TableCell;
+import javafx.scene.control.TreeTableCell;
+import javafx.scene.control.skin.LabelSkin;
+import javafx.scene.control.skin.TableCellSkin;
+import javafx.scene.control.skin.TreeTableCellSkin;
+import org.junit.jupiter.api.Test;
+import com.sun.javafx.scene.layout.RegionHelper;
+import com.sun.javafx.tk.Toolkit;
+
+/**
+ * Tests textTruncated property of Labeled, using Label, TableCell, and TreeTableCell controls
+ * (the last two contain conditional code that redirects the execution of computePrefWidth())
+ * in their skins to different code paths.
+ */
+public class LabeledTruncatedTest {
```

Comment:
It might be worth adding a test for `Button`.

Reply:
test added